argh/src/lib.rs

```diff
@@ -455,6 +460,23 @@ impl<T: SubCommand> SubCommands for T {
     const COMMANDS: &'static [&'static CommandInfo] = &[T::COMMAND];
 }
 
+/// Trait implemented by values returned from a dynamic subcommand handler.
+pub trait DynamicSubCommand: Sized {
```

Comment:
Not sure what thread you're referring to. Don't immediately see how this improves the design.

Reply:
Drat, my comment got lost, let me try to revive it.

So what I was wondering was centered around if an impl of SubCommands would expect us to potentially have multiple instances of `DynamicSubCommand`. 

As an alternative, we could instead only allow for a single `DynamicSubCommand`, and then that command would be responsible for potentially merging together multiple dynamic subcommands. In order to implement this though, we'd need a way to merge together the multiple command lists from the downstream dynamic subommands. Furthermore, we'd have to change `DynamicSubcommand::commands()` to be able to return a dynamic list (or use something like once_cell to handle this).

Either way, we should have tests for this scenario, and it should be documented.

Reply:
I don't know what is meant exactly by "downstream dynamic subcommands". DynamicSubCommand is deliberately the end of what we can infer about the structure of argument parsing. In terms of the commands it gives us we're already using the once_cell approach, so that's tested.

We do allow more than one at this point and it doesn't seem to break anything. It's not my envisioned usage pattern but I don't know if there's good reason to prevent it either?

Reply:
Looking again, while we allowed more than one, it didn't actually work correctly. I've pivoted to only allowing one dynamic variant. This also saves us the question of "what if two dynamic subcommands can take the same command name". We do however still have the question of "what if a dynamic subcommand tries to squat on the name of a static subcommand". Still thinking about that one. I think our tests are adequate modulo that, unless we have compile-fail tests where we could check that multiple dynamic commands prevents compiling? I see some TODOs suggesting we want such tests but I don't know if we actually have them.

What should documentation look like?